reader-gtfs/src/test/java/com/graphhopper/gtfs/fare/FareTest.java

```diff
@@ -23,168 +23,190 @@
-
-        twoLegsWithDistinctZones = new Trip();
-        twoLegsWithDistinctZones.segments.add(new Trip.Segment("only_feed_id", "Route_1",0, "S1", "S4", new HashSet<>(Arrays.asList("1"))));
-        twoLegsWithDistinctZones.segments.add(new Trip.Segment("only_feed_id", "Route_2",5000, "S4", "S1", new HashSet<>(Arrays.asList("2"))));
-        twoLegsWithDistinctZones.segments.add(new Trip.Segment("only_feed_id", "Route_1",6000, "S1", "S4", new HashSet<>(Arrays.asList("3"))));
-
-        twoLegsWithDistinctFeeds = new Trip();
-        twoLegsWithDistinctFeeds.segments.add(new Trip.Segment("only_feed_id", "Route_1",0, "S1", "S4", new HashSet<>()));
-        twoLegsWithDistinctFeeds.segments.add(new Trip.Segment("feed_id_2", "Route_2",5000, "T", "T", new HashSet<>()));
+        Trip trip;
+        trips.add(trip = new Trip());
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 0, "S1", "S2", new HashSet<>(Arrays.asList("1", "2", "3"))));
+
+        trips.add(trip = new Trip());
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 0, "S1", "S4", new HashSet<>(Arrays.asList("1"))));
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_2", 6000, "S4", "S1", new HashSet<>(Arrays.asList("1"))));
+
+        trips.add(trip = new Trip());
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 0, "S1", "S4", new HashSet<>(Arrays.asList("2", "3"))));
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_2", 5000, "S4", "S1", new HashSet<>(Arrays.asList("2", "3"))));
+
+        trips.add(trip = new Trip());
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 0, "S1", "S4", new HashSet<>(Arrays.asList("1"))));
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_2", 5000, "S4", "S1", new HashSet<>(Arrays.asList("2"))));
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 6000, "S1", "S4", new HashSet<>(Arrays.asList("3"))));
+
+        trips.add(trip = new Trip());
+        trip.segments.add(new Trip.Segment("only_feed_id", "Route_1", 0, "S1", "S4", new HashSet<>()));
+        trip.segments.add(new Trip.Segment("feed_id_2", "Route_2", 5000, "T", "T", new HashSet<>()));
+    }
 
+    private static class DataPointProvider implements ArgumentsProvider {
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            List<Object[]> dataPoints = new ArrayList<>();
+            for (int i = 0; i < fares.size(); i++) {
+                Map<String, Map<String, Fare>> fare = fares.get(i);
+                for (int j = 0; j < trips.size(); j++) {
+                    Trip trip = trips.get(j);
+                    dataPoints.add(new Object[]{fare, trip, "fare " + i + ", trip: " + j});
+                }
+            }
+            return dataPoints.stream().map(Arguments::of);
```

Comment:
This is just a cartesian product of all trips/fares and each tests will reject some of themm using `Assumptions`. But maybe we can also create multiple such providers that filter out certain trip/fare data points already here.